core.ts

```diff
@@ -54,14 +54,17 @@ const isObject = (v: any): v is object =>
 const isPromiseRejector = (obj: any): obj is PinejsClientCoreFactory.PromiseRejector => {
 	return obj != null && obj.reject != null
 }
-const isValidOption = (key: string): key is keyof PinejsClientCoreFactory.ODataOptions => {
+const isValidOption = (key: string | number): key is keyof PinejsClientCoreFactory.ODataOptions => {
```

Comment:
Why is number allowed?

Reply:
It's a requirement in typescript 2.9+, they changed the valid keyof types I think and you have to allow all possible key types in a function that asserts keyof

Reply:
Since ODataOptions don't actually have numeric props, how about doing `(key: string): key is keyof PinejsClientCoreFactory.ODataOptions & string` instead, which is also backwards compatible with TS < 2.9 and is how the code was working before as well.

PS: we could alternatively do `key is Extract<keyof PinejsClientCoreFactory.ODataOptions, string>` back that forces our consumers to TS 2.8+

Reply:
The latter is a breaking change though, so it would warrant a major version bump for pine-client.

Reply:
Yea, that's why I proposed the first one :)
I should have made clear that I was also not voting in favor of the PS (which is the TS proposed way of doing it) since I had the same concerns that you described in mind.
See: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-9.html

Reply:
This function doesn't get exposed at all so we can make any changes we want to it without it being a breaking change, since a user will not interact with it or its types directly